### PR TITLE
Add rich text editor for showcase content

### PR DESCRIPTION
**Pull request by pdelboca**
This PR adds the option to use CKEditor as wysiwyg editor for Showcase's content.

image

image

CKEditor is configured with [Markdown output](https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html) plugin to write the content as markdown text.

This PR also includes the webpacks config files to to build our own CKEditor source file to be used by the extension as described in the [official docs](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/advanced-setup.html#scenario-2-building-from-source) of the editor.

### TODO:
- [x] Customize the menu buttons of the wysiwyg editor
- [x] Fix languaje of the source file created by webpacks
- [x] Add tests
- [x] Update README.md file

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -79,6 +79,20 @@ do::
     pip install -r dev-requirements.txt
 
 
+The extension contains a custom build of CKEditor to allow using a WYSIWYG editor
+to write the content of the showcase. I has been built using `webpack` and the
```
Comment (amercader) [inline README.rst:83]:
```suggestion
to write the content of the showcase. It has been built using `webpack` and the
```

**Comment by amercader** [comment]
Really well done @pdelboca . One tiny improvement if you have time, it would be good to increase the height of the control initially so it's clearer that you can input a long description:

Screenshot_2020-05-12 Create Showcase - CKAN

**Comment by TkTech** [comment]
This looks great!

There are several extensions now with WYSIWYG editors. I'm concerned that they all seem to be bundling it (usually ckeditor), and it often doesn't get updated it until something breaks.

It may be better to offer an editor component as a core ckan extension for other extensions to use, rather than bundling their own, so that the end-user experience is consistent. It would make adding ckeditor plugins consistent, or enable swapping out ckeditor for tinymce. showcase-ckeditor.js for example provides no way to add additional configuration, or plumbing to log errors.

**Comment by pdelboca** [comment]
We were discussing the same @TkTech, I can probably implement a core extension as you suggest (or a CKAN javascript module) with this logic to benefit all the ecosystem of extensions. This workflow using `webpack` is really nice and also worth discussing as a good practice for extension's development. 

I will suggest a PR/Improvement to CKAN to discuss it further.

**Pull request merged**